### PR TITLE
Feature add oauth

### PR DESCRIPTION
**Pull request by JirawadeeSampusri**
I have already done Oauth google but the secret key of Oauth can't be hidden and I also try to use Config but it doesn't work.
For the Logout button is on the profile page.

### CODE REVIEW

File: Kvent/forms.py [hunk 1]
```diff
@@ -44,4 +44,5 @@ class Meta(UserCreationForm.Meta):
             'username': forms.TextInput(attrs={'class': 'form-control', 'placeholder': 'Username'}),
             'password1': forms.PasswordInput(attrs={'class': 'form-control', 'placeholder': 'Password'}),
             'password2': forms.PasswordInput(attrs={'class': 'form-control', 'placeholder': 'Password Again'}),
+
```
Comment (bleachjade) [inline Kvent/forms.py:51]: delete this

File: Kvent/templates/kvent/profile.html [hunk 2]
```diff
@@ -19,9 +19,14 @@
         <ul>
             <li><a href="{% url 'index' %}">Kvent</a></li>
             <li><a href="{% url 'index' %}" id="event-logo">Events</a></li>
+            {% if user.is_authenticated %}
+                <a class="btn btn-primary" href="{% url 'logout' %}">Logout</a>
+            {% endif %}
```
Comment (bleachjade) [inline Kvent/templates/kvent/profile.html]: You should also add this button in the index and don't forget to parse it inside `<li>` tag

File: Kvent/templates/kvent/profile.html [hunk 3]
```diff
@@ -19,9 +19,14 @@
...
         </ul>
     </nav>
 
+
+
```
Comment (bleachjade) [inline Kvent/templates/kvent/profile.html]: delete unnecessary space

File: Kvent/templates/registration/createaccount.html [hunk 4]
```diff
@@ -60,3 +60,13 @@ <h2>Sign up</h2>
...
+
+<!-- multistep form -->
+<form id="msform">
+  <!-- progressbar -->
+  <ul id="progressbar">
+    <li class="active">Account Setup</li>
+    <li>Social Profiles</li>
+    <li>Personal Details</li>
+  </ul>
+
```
Comment (bleachjade) [inline Kvent/templates/registration/createaccount.html]: What is this used for? if it did not use in any of our code, please delete it

File: Kvent/templates/registration/login.html [hunk 5]
```diff
@@ -38,11 +38,20 @@
...
       <button type="submit" value="{{ next }}" >Login</button>
     </form>
+     <p class="forgot" align="center"><a>Don't have any account? </p>
+    <br>
+    <p class="forgot1" align="center" ><a href="{% url 'signup' %}"> Create one </a></p> 
+    </form>
+    
+     <a class="btn btn-primary" href="{% url 'social:begin' 'google-oauth2' %}">
+        <p1 class="google">Google+</p1>
+     </a>
```
Comment (bleachjade) [inline Kvent/templates/registration/login.html]: keep this button but the rest of it just use the template in the develop branch

**Review by bleachjade — CHANGES_REQUESTED** [review]

File: mysite/settings.py [hunk 6]
```diff
@@ -23,6 +24,8 @@
 
 # SECURITY WARNING: keep the secret key used in production secret!
 SECRET_KEY = config('SECRET_KEY', default='secret-key')
+SOCIAL_AUTH_GOOGLE_OAUTH2_KEY = '258892222487-e3lapeoqq0tem2k5oi21asccebpgv4cv.apps.googleusercontent.com' 
+SOCIAL_AUTH_GOOGLE_OAUTH2_SECRET = 'QjQaLqQFiP_Hmgfv3lPCr9sJ'
```
Comment (bleachjade) [inline mysite/settings.py]: please try to add it in .env and use config, we should not push the secret key to our repo

File: mysite/settings.py [hunk 7]
```diff
@@ -133,6 +142,15 @@
 STATICFILES_DIRS = (os.path.join(BASE_DIR, 'static'), )
 STATIC_ROOT = os.path.join(BASE_DIR, 'staticfiles')
 
+STATIC_URL = '/static/'
```
Comment (bleachjade) [inline mysite/settings.py]: delete this, we already have a STATIC_URL above

File: mysite/urls.py [hunk 8]
```diff
@@ -15,9 +15,18 @@
...
     path('', include('Kvent.urls')),
     path('admin/', admin.site.urls),
+    # path(
+    #     'logout/',
+    #     LogoutView.as_view(template_name=settings.LOGOUT_REDIRECT_URL),
+    #     name='logout'
+    # ),
+    # path('manage/', views.manage, name='manage'),
+    # path('', include('social_django.urls', namespace='social')),
     # path('accounts/', include('django.contrib.auth.urls')),
```
Comment (bleachjade) [inline mysite/urls.py]: delete comment line out

**Review by bleachjade — CHANGES_REQUESTED** [review]

**Review by bleachjade — CHANGES_REQUESTED** [review]
**Don't include the files in the api folder!** When you doing a google auth, you just add auth things in the setting not creating a new manage.py and others file

File: Kvent/templates/kvent/profile.html [hunk 9]
```diff
@@ -19,15 +19,16 @@
         <ul>
             <li><a href="{% url 'index' %}">Kvent</a></li>
             <li><a href="{% url 'index' %}" id="event-logo">Events</a></li>
+            {% if user.is_authenticated %}
+                <a class="btn btn-primary" href="{% url 'logout' %}">Logout</a>
```
Comment (forfeen) [inline Kvent/templates/kvent/profile.html]: Please place the logout button at the right of profile page.

**Pull request merged**